### PR TITLE
Gating Unsupported runtime

### PR DESCRIPTION
**Pull request by jonife**
#### Why is this change necessary?
Help in gating runtimes SAMCLI is yet to supported to be accessed by customer.

#### How does it address the issue?
Removes runtime from list of runtimes generated from the manifest file in CLI template repo

#### What side effects does this change have?
None

#### Mandatory Checklist
**PRs will only be reviewed after checklist is complete**

- [x] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document if needed ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [x] Write/update unit tests
- [ ] Write/update integration tests
- [ ] Write/update functional tests if needed
- [x] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: samcli/commands/init/interactive_init_flow.py [hunk 1]
```diff
@@ -349,6 +350,20 @@ def get_sorted_runtimes(runtime_option_list):
     return sorted_runtime
 
 
+def get_supported_runtime(runtime_list):
```
Comment (hawflau) [inline samcli/commands/init/interactive_init_flow.py:353]: Please add docstring. Also, can we add a unit test for this function?

File: samcli/commands/init/interactive_init_flow.py [hunk 2]
```diff
@@ -349,6 +350,20 @@ def get_sorted_runtimes(runtime_option_list):
...
+    supported_runtime_list = []
+    error_message = ""
+    for runtime in runtime_list:
+        if runtime not in INIT_RUNTIMES:
+            if not error_message:
```
Comment (hawflau) [inline samcli/commands/init/interactive_init_flow.py:358]: If there are multiple not-supported runtimes, it seems the debug message will still be shown repeatedly?

**Review by hawflau — CHANGES_REQUESTED** [review]

**Review by hawflau — APPROVED** [review]
LGTM!

File: samcli/commands/init/interactive_init_flow.py [hunk 3]
```diff
@@ -300,6 +300,9 @@ def _get_choice_from_options(chosen, options, question, msg):
     click_choices = []
 
     options_list = options if isinstance(options, list) else list(options.keys())
+    options_list = (
+        get_sorted_runtimes(options_list) if msg == "Runtime" and not isinstance(options, list) else options_list
```
Comment (torresxb1) [inline samcli/commands/init/interactive_init_flow.py:304]: why do we need the condition `not isinstance(options, list)`?

File: samcli/commands/init/interactive_init_flow.py [hunk 4]
```diff
@@ -338,16 +339,45 @@ def get_sorted_runtimes(runtime_option_list):
...
+    error_message = ""
+    for runtime in runtime_list:
+        if runtime not in INIT_RUNTIMES:
+            if not error_message:
+                error_message = "Additional runtimes may be available in the latest SAM CLI version. \
+                    Upgrade your SAM CLI to see the full list."
+                LOG.debug(error_message)
+            continue
+        supported_runtime_list.append(runtime)
+    return supported_runtime_list
```
Comment (torresxb1) [inline samcli/commands/init/interactive_init_flow.py:378]: Shorter way of doing this
```suggestion
    supported_runtime_list = list(set(runtime) & set(INIT_RUNTIMES))
    if len(supported_runtime_list) != len(runtime):
        LOG.debug(
                    "Additional runtimes may be available in the latest SAM CLI version. Upgrade your SAM CLI to see the full list."
                )
    return supported_runtime_list
```

Comment (jonife) [reply samcli/commands/init/interactive_init_flow.py:378]: This doesn't covers an edge where we INIT_runtime doesn't all the runtime, ie custom runtime support.

Comment (torresxb1) [reply samcli/commands/init/interactive_init_flow.py:376]: but if INIT_runtime doesn't contain all the runtimes, it would throw an error anyway because of the check `if runtime not in INIT_RUNTIMES`, no? (I believe the `supported_runtime_list.append(runtime)` wouldn't be run because of the `continue` before it). 
Also, if that's a valid edge case, could we add a test to cover it?

Comment (torresxb1) [reply samcli/commands/init/interactive_init_flow.py:376]: if you need to return all of the runtimes in `runtime_list` then you could just update my suggestion to return `runtime_list` instead of `supported_runtime_list`. But I'm not sure if I understand correctly
